src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.ts

```diff
@@ -1,21 +1,43 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
 import { ResourceService } from './../../services';
 import { ActivatedRoute, Router, NavigationExtras } from '@angular/router';
+import * as _ from 'lodash';
 
 @Component({
   selector: 'app-collection-player-metadata',
   templateUrl: './collection-player-metadata.component.html',
   styleUrls: ['./collection-player-metadata.component.css']
 })
 export class CollectionPlayerMetadataComponent implements OnInit {
-  @Input() metaData: object;
+  @Input() metaData: any;
   public collectionMeta: any = {};
   public collectionId: string;
+  contributions: string;
+  contributionsLength: number;
+  showContentCreditsModal: boolean;
+  contentCreditsData: any = {};
+
   constructor(public resourceService: ResourceService, private activatedRoute: ActivatedRoute) { }
 
   ngOnInit() {
     this.activatedRoute.params.subscribe((params) => {
       this.collectionId = params.collectionId;
     });
   }
+
+  showContentCredits () {
```

Comment:
Check review comments and if same code written in two components it should be moved to service to not get into **DRY** code anti pattern